credit.go

```diff
@@ -0,0 +1,255 @@
+	// a debit or credit adjustment.
+	DebitCreditIndicator string `json:"debitCreditIndicator,omitempty"`
+	// reserved is a field reserved for future use.  Reserved should be blank.
+	reserved string
+	// validator is composed for image cash letter data validation
+	validator
+	// converters is composed for image cash letter to golang Converters
+	converters
+}
+
+// NewCredit returns a new credit with default values for non exported fields
+func NewCredit() *Credit {
+	cr := &Credit{}
+	cr.setRecordType()
+	return cr
+}
+
+func (cr *Credit) setRecordType() {
+	if cr == nil {
+		return
+	}
+	cr.recordType = "61"
+	cr.reserved = "   "
+}
+
+// Parse takes the input record string and parses the BundleControl values
+func (cr *Credit) Parse(record string) {
+	if utf8.RuneCountInString(record) < 56 {
+		return
+	}
+
+	// Character position 1-2, Always "61"
+	cr.setRecordType()
+	// 03–17
+	cr.AuxiliaryOnUs = cr.parseStringField(record[2:17])
+	// 18
+	cr.ExternalProcessingCode = cr.parseStringField(record[17:18])
+	// 19–27
+	cr.PayorBankRoutingNumber = cr.parseStringField(record[18:27])
+	// 28–47
+	cr.CreditAccountNumberOnUs = cr.parseStringField(record[27:47])
+	// 48–57
+	cr.ItemAmount = cr.parseNumField(record[47:57])
+	// 58–72
+	cr.ECEInstitutionItemSequenceNumber = cr.parseStringField(record[57:72])
+	// 73
+	cr.DocumentationTypeIndicator = cr.parseStringField(record[72:73])
+	// 74
+	cr.TypeAccountCode = cr.parseStringField(record[73:74])
+	// 75
+	cr.SourceWorkCode = cr.parseStringField(record[74:75])
+	// 76
+	cr.WorkType = cr.parseStringField(record[75:76])
+	// 77
+	cr.DebitCreditIndicator = cr.parseStringField(record[76:77])
+	// 78–80
+	cr.reserved = "   "
+
+}
+
+func (cr *Credit) UnmarshalJSON(data []byte) error {
+	type Alias Credit
+	aux := struct {
+		*Alias
+	}{
+		(*Alias)(cr),
+	}
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+	cr.setRecordType()
+	return nil
+}
+
+// String writes the BundleControl struct to a string.
+func (cr *Credit) String() string {
+	var buf strings.Builder
+	buf.Grow(80)
+	buf.WriteString(cr.recordType)
+	buf.WriteString(cr.AuxiliaryOnUsField())
+	buf.WriteString(cr.ExternalProcessingCodeField())
+	buf.WriteString(cr.PayorBankRoutingNumberField())
+	buf.WriteString(cr.CreditAccountNumberOnUsField())
+	buf.WriteString(cr.ItemAmountField())
+	buf.WriteString(cr.ECEInstitutionItemSequenceNumberField())
+	buf.WriteString(cr.DocumentationTypeIndicatorField())
+	buf.WriteString(cr.TypeAccountCodeField())
+	buf.WriteString(cr.SourceWorkCodeField())
+	buf.WriteString(cr.WorkTypeField())
+	buf.WriteString(cr.DebitCreditIndicatorField())
+	buf.WriteString(cr.reservedField())
+	return buf.String()
+}
+
+// Validate performs image cash letter format rule checks on the record and returns an error if not Validated
+// The first error encountered is returned and stops the parsing.
+func (cr *Credit) Validate() error {
```

Comment:
The PDF seems to suggest that `ECEInstitutionItemSequenceNumber` is mandatory, but the "comment" column says "Spaces"... Should that be enforced here?

Reply:
ECEInstitutionItemSequenceNumber used "Spaces" in some sample codes
so changed into omitempty field although require field

Reply:
This case is handled in other records by using the `ECEInstitutionItemSquenceNumberField()` function. You could do something like this to ensure the field gets validated:
```go
	if cr.ECEInstitutionItemSequenceNumberField() != "               "  {
		return &FieldError{FieldName: "ECEInstitutionItemSequenceNumber", Value: cr.ECEInstitutionItemSequenceNumber, Msg: msgInvalid}
```